src/Algebra/Linear/Properties.agda

```diff
@@ -0,0 +1,180 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear algebraic structures.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra
+  using (CommutativeRing; Congruent₁; Congruent₂; Op₁; Op₂)
+open import Algebra.Module        using (Module)
+open import Algebra.Linear.Bundles
+open import Level                 using (Level; _⊔_; suc)
+
+module Algebra.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  {mod       : Module ring m ℓm}
+  (vs        : VectorSpace mod)
+  where
+
+open import Algebra.Module.Construct.TensorUnit using (⟨module⟩)
+open import Algebra.Module.Morphism.Bundles
+import      Algebra.Module.Morphism.ModuleHomomorphism as ModHomo
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+open import Function
+open import Relation.Binary
+import      Relation.Binary.ExtensionalEquivalence     as ExtEq
+import      Relation.Binary.PropositionalEquality      as Eq
+open import Relation.Binary.Reasoning.MultiSetoid
+
+open VectorSpace vs
+open ExtEq       setoid
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+-- Some consequences of certain `VectorSpace` property fields.
+v∙g[x]+y-cong₂ : {g : V → V} {v x w : V} {y z : S} →
+                 Congruent _≈ᴹ_ _≈ᴹ_ g → x ≈ᴹ w → y ≈ z →
+                 v ∙ g x + y ≈ v ∙ g w + z
+v∙g[x]+y-cong₂ {g} {v} {x} {w} {y} {z} g-cong x≈w y≈z = begin⟨ setoid ⟩
+  v ∙ g x + y ≈⟨ +-congʳ (∙-congˡ (g-cong x≈w)) ⟩
+  v ∙ g w + y ≈⟨ +-congˡ y≈z ⟩
+  v ∙ g w + z ∎
+
+foldr-cong : ∀ {f g : V → S → S} {d e : S} →
+             (∀ {y z} → y ≈ z → ∀ x → f x y ≈ g x z) → d ≈ e →
+             foldr f d ≗ foldr g e
+foldr-cong f≈g d≈e []       = d≈e
+foldr-cong f≈g d≈e (x ∷ xs) = f≈g (foldr-cong f≈g d≈e xs) x
```

Comment:
I feel this should be a far more general proof in `Data.List.Properties`? (if it doesn't already exist!)

Reply:
Yeah, I get that. But, the one in `Data.List.Properties` uses _equality_, as opposed to _equivalence_, which I can't use here:

```agda
foldr-cong : ∀ {f g : A → B → B} {d e : B} →
             (∀ x y → f x y ≡ g x y) → d ≡ e →
             foldr f d ≗ foldr g e
```

That's why I had to write my own. :(

Reply:
Would `Data.List.Relation.Binary.Setoid.foldr⁺` be useful here?

Reply:
> Would `Data.List.Relation.Binary.Setoid.foldr⁺` be useful here?

I don't think so, only because it applies to `Op2 A` while mine applies to `V -> S -> S`.